modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/portlet/action/ImportSystemDataSetMVCResourceCommand.java

```diff
@@ -87,10 +103,205 @@ protected void doTransactionalCommand(
+					).put(
+						"title_i18n",
+						() -> _getLocalizeableValue(
+							defaultLanguageId,
+							_getOptionalValue(data.get("title")))
+					).build());
+			}
+
+			_objectEntryService.addObjectEntry(
+				0, objectDefinitionId, objectEntryValues, new ServiceContext());
+		}
+	}
+
+	private void _addItemDataSetActionObjectEntries(
+			long dataSetId, String defaultLanguageId,
+			FDSItemActionList fdsItemActionList,
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse, long objectDefinitionId)
+		throws Exception {
+
+		List<FDSActionDropdownItem> fdsActionDropdownItems =
+			fdsItemActionList.getFDSActionDropdownItems(
+				httpServletRequest, httpServletResponse);
+
+		for (FDSActionDropdownItem fdsActionDropdownItem :
+				fdsActionDropdownItems) {
+
+			Map<String, Object> data =
+				(Map<String, Object>)fdsActionDropdownItem.get("data");
+
+			_objectEntryService.addObjectEntry(
+				0, objectDefinitionId,
+				HashMapBuilder.<String, Serializable>put(
+					"confirmationMessage_i18n",
+					() -> _getLocalizeableValue(
+						defaultLanguageId,
+						_getOptionalValue(data.get("confirmationMessage")))
+				).put(
+					"confirmationMessageType",
+					() -> _getOptionalValue(data.get("confirmationMessageType"))
+				).put(
+					"errorMessage_i18n",
+					() -> _getLocalizeableValue(
+						defaultLanguageId,
+						_getOptionalValue(data.get("errorMessage")))
+				).put(
+					"icon",
+					() -> _getOptionalValue(fdsActionDropdownItem.get("icon"))
+				).put(
+					"label_i18n",
+					() -> _getLocalizeableValue(
+						defaultLanguageId,
+						_getOptionalValue(fdsActionDropdownItem.get("label")))
+				).put(
+					"method", () -> _getOptionalValue(data.get("method"))
+				).put(
+					"modalSize", () -> _getOptionalValue(data.get("modalSize"))
+				).put(
+					"permissionKey",
+					() -> _getOptionalValue(data.get("permissionKey"))
+				).put(
+					"r_dataSetToDataSetActions_l_dataSetId", dataSetId
+				).put(
+					"requestBody",
+					() -> _getOptionalValue(data.get("requestBody"))
+				).put(
+					"successMessage_i18n",
+					() -> _getLocalizeableValue(
+						defaultLanguageId,
+						_getOptionalValue(data.get("successMessage")))
+				).put(
+					"target",
+					String.valueOf(fdsActionDropdownItem.get("target"))
+				).put(
+					"title_i18n",
+					() -> _getLocalizeableValue(
+						defaultLanguageId, _getOptionalValue(data.get("title")))
+				).put(
+					"type", "item"
+				).put(
+					"url",
+					() -> _getOptionalValue(fdsActionDropdownItem.get("href"))
+				).build(),
+				new ServiceContext());
+		}
+	}
+
+	private Serializable _getLocalizeableValue(
+		String languageId, Object value) {
+
+		if (value == null) {
+			return null;
+		}
+
+		return HashMapBuilder.put(
+			languageId, String.valueOf(value)
+		).build();
+	}
+
+	private Serializable _getOptionalValue(Object value) {
+		if (value == null) {
+			return null;
+		}
+
+		return String.valueOf(value);
+	}
```

Comment:
This, along with lambdas, is a mechanism that ensures we don't write `null` strings on unset optional values. There might be a util that could be used here, but I couldn't find one.